### PR TITLE
Refactored FilterListService

### LINKED ISSUE #1137: Clean up the `BaseListFilterService`
Since #1132 is merged, the `BaseListFilterService` injects the `HistoryService` which is only relevant for meetings. The `HistoryService` (and maybe the `StorageService`, too) should be extracted from the `BaseListFilterService`.

### PR DESCRIPTION
**Pull request by luisa-beerboom**
Closes #1137

### CODE REVIEW

File: client/src/app/ui/base/filter-service.ts [hunk 1]
```diff
@@ -51,14 +51,107 @@ type OsFilterOptionCondition = OsFilterOptionConditionType | OsFilterOptionCondi
...
+     */
+    filterStack: OsFilterIndicator<V>[];
+
+    /**
+     * The key to access stored valued
+     */
+    storageKey: string;
+}
+
+export class FilterListData<V>{
```
Comment (GabrielInTheWorld) [inline client/src/app/ui/base/filter-service.ts:100]: The implementation of `FilterListData` should implement the interface.

File: client/src/app/site/pages/meetings/base/base-meeting-filter-list.service.ts [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+        }
+
+        if (this.filterListData.inputDataSubscription) {
+            this.filterListData.inputDataSubscription.unsubscribe();
+            this.filterListData.inputDataSubscription = null;
+        }
+        this.filterListData.inputDataSubscription = inputData.subscribe(data => {
+            this.filterListData.inputData = data;
+            this.updateFilteredData();
+        });
```
Comment (GabrielInTheWorld) [inline client/src/app/site/pages/meetings/base/base-meeting-filter-list.service.ts:149]: This is duplicated code from the `BaseFilterListService`

File: client/src/app/site/pages/meetings/base/base-meeting-filter-list.service.ts [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+    // }
+
+    // public set inputData(data: V[]) {
+    //     this.baseFilterListService.inputData = data;
+    // }
+
+    /**
+     * The key to access stored valued
+     */
+    // protected abstract readonly storageKey: string;
```
Comment (GabrielInTheWorld) [inline client/src/app/site/pages/meetings/base/base-meeting-filter-list.service.ts:104]: Remove

File: client/src/app/site/base/base-filter.service/base-filter-list.service.ts [hunk 4]
```diff
@@ -90,63 +90,74 @@ export abstract class BaseFilterListService<V extends BaseViewModel> implements
...
 
     /**
      * The key to access stored valued
      */
-    protected abstract readonly storageKey: string;
+    // protected abstract readonly storageKey: string;
 
-    public constructor(private store: StorageService, private historyService: HistoryService) {}
+    public constructor(private store: StorageService) {
+        this.filterListData = new FilterListData<V>();
```
Comment (GabrielInTheWorld) [inline client/src/app/site/base/base-filter.service/base-filter-list.service.ts:113]: The idea behind the creating of an instance of `FilterListData` was not to just use it as data wrapper. Instead, please implement a real `create` or `build` function, which returns an instance of `FilterListData`. The `FilterListData` instance should then provide every function, which works on its data. The `BaseFilterListService` should provide functions, which filters a source by given `FilterDefinition`s.

**Review by GabrielInTheWorld — CHANGES_REQUESTED** [review]
I haven't reviewed every yet, but there are already some things which should be changed.

**Comment by luisa-beerboom** [comment]
More recent and practical solution in #1162

**Pull request closed without merging**